dot/types/body_extrinsics.go

```diff
@@ -23,33 +23,42 @@ import (
 
 	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/lib/common/optional"
-	"github.com/ChainSafe/gossamer/lib/scale"
-	scale2 "github.com/ChainSafe/gossamer/pkg/scale"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
-// Body is the encoded extrinsics inside a state block
-type Body []byte
+// Body is the extrinsics(not encoded) inside a state block.
+type Body []Extrinsic
 
-// NewBody returns a Body from a byte array
-func NewBody(b []byte) *Body {
-	body := Body(b)
+// NewBody returns a BodyExtrinsics from an Extrinsic array.
+func NewBody(e []Extrinsic) *Body {
```

Comment:
Any reason for the body to be a pointer? Since it is already a slice.

Reply:
I have kept what was being used previously. 
Not using pointer would be safer in general. Changing this from pointer, would mean edit this at a lot places. But, I think doing so would be the right way. Will change it.

Reply:
At most place where body is returned it is returned as a pointer.

So if passed without pointer to the slice (`Body` and not `*Body`), you can modify it, but to an extent. From what I read, you can modify, values but not length of the slice. If you try to append to it, slice won't have values that you expected. So far, I have not seen anywhere `Body` getting modified.

I also see that at places a pointer to Body and a pointer is stored for even other slices.
```
type BlockData struct {
	Hash          common.Hash
	Header        *Header
	Body          *Body
	Receipt       *[]byte
	MessageQueue  *[]byte
	Justification *[]byte
}
```
@noot is returning or storing `*Body` (and also other slices, like `MessageQueue *[]byte`) done deliberately? Should I care returning `Body` instead of `*Body`?

Reply:
all the fields in the BlockData should be a pointer as they are optional types. please leave this as is

Reply:
actually, for this functions specifically, I'm not sure if it really matters. I think it could be either